.github/workflows/styles.yml

```diff
@@ -0,0 +1,82 @@
+name: Styles
+on:
+  pull_request:
+    paths:
+      - .github/workflows/styles.yml
+      - data/resources*
+      - data/styles*
+      - data/drules*
+      - data/mapcss*
+      - skin_generator/
+      - tools/kothic/
+      - tools/python/stylesheet/
+      - tools/unix/generate_drules.sh
+      - tools/unix/generate_styles.sh
+      - tools/unix/generate_symbols.sh
+
+jobs:
+  check_styles:
+    name: Styles up to date
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Restore Boost submodule from cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            3party/boost
+            .git/modules/3party/boost
+          key: boost-submodule
+
+      - name: Parallel submodules checkout
+        shell: bash
+        run: git submodule update --depth 1 --init --recursive --jobs=$(($(nproc) * 20))
+
+      - name: Install dependencies
+        run: |
+          sudo apt update -y
+          sudo apt install -y \
+            optipng \
+            libharfbuzz-dev \
+            libxrandr-dev \
+            libxinerama-dev \
+            libxcursor-dev \
+            libxi-dev \
+            qt6-base-dev \
+            libqt6svg6-dev \
+            qt6-positioning-dev \
+            libqt6positioning6-plugins \
+            libqt6positioning6
+          
+          python -m pip install --upgrade pip
+          pip install protobuf==3.20.3
```

Comment:
A common approach for python projects is to have `requirements.txt` file with all dependencies and run `pip install -r requirements.txt`. It will make easier for developers to find out which protobuf version is required.

Reply:
and we have this requirements.txt. But it is broken :)
the protobuf version specified in this file doesn't exists

Reply:
Would it be better to fix it instead of duplicating environments here and there?